File: sprint2/C++/lesson7code1.cpp
#include <iostream>
#include <vector>
#include <string>

class Stack {
private:
    std::vector<std::string> items;

public:
    void push(std::string item) {
        items.push_back(item);
    }

    std::string pop() {
        std::string lastItem = items.back();
        items.pop_back();
        return lastItem;
    }

    std::string peek() {
        return items.back();
    }

    int size() {
        return items.size();
    }
};

int main() {
    Stack stack;
    stack.push("apple");
    stack.push("banana");
    stack.push("orange");
    stack.pop();

    return 0;
}

File: sprint2/go/lesson7code1.go
type Stack struct {
	items []string
}

func NewStack() *Stack {
	return &Stack{items: []string{}}
}

func (s *Stack) Push(item string) {
	s.items = append(s.items, item)
}

func (s *Stack) Pop() string {
	lastIndex := len(s.items) - 1
	lastItem := s.items[lastIndex]
	s.items = s.items[:lastIndex]
	return lastItem
}

func (s *Stack) Peek() string {
	lastIndex := len(s.items) - 1
	return s.items[lastIndex]
}

func (s *Stack) Size() int {
	return len(s.items)
}

stack := NewStack()
stack.Push("apple")
stack.Push("banana")
stack.Push("orange")
stack.Pop()
